pkg/service/service.go

```diff
@@ -662,6 +706,108 @@ func (s *service) ListModels(ctx context.Context, pageSize int32, pageToken stri
 	return pbModels, int32(totalSize), nextPageToken, err
 }
 
+func (s *service) ListModelTriggers(ctx context.Context, req *modelpb.ListModelTriggersRequest, view modelpb.View) (*modelpb.ListModelTriggersResponse, error) {
+	pageSize := s.pageSizeInRange(req.GetPageSize())
+	page := s.pageInRange(req.GetPage())
+
+	orderBy, err := ordering.ParseOrderBy(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var startTimeFrom, startTimeTo *time.Time
+	if req.StartTimeFrom.IsValid() {
+		t := req.GetStartTimeFrom().AsTime()
+		startTimeFrom = &t
+	}
+	if req.StartTimeTo.IsValid() {
+		t := req.GetStartTimeTo().AsTime()
+		startTimeTo = &t
+	}
+	logger, _ := custom_logger.GetZapLogger(ctx)
+
+	triggers, totalSize, err := s.repository.ListModelTriggers(ctx, int64(pageSize), int64(page), orderBy, req.ModelUid, startTimeFrom, startTimeTo)
+	if err != nil {
+		return nil, err
+	}
+
+	metadataMap := make(map[string][]byte)
+	if view == modelpb.View_VIEW_FULL {
```

Comment:
I feel there's plenty of code dedicated to the full view but we'll never list with that option. I'm wondering if it'd be better to force the basic view on the list endpoint and have a simpler function.

Reply:
I think so too. We could create another ticket for that